presto-parquet/src/main/java/io/prestosql/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -78,32 +92,77 @@ protected boolean isValueNull()
         return ParquetTypeUtils.isValueNull(columnDescriptor.isRequired(), definitionLevel, columnDescriptor.getMaxDefinitionLevel());
     }
 
-    public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor)
+    public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor, Type prestoType)
```

Comment:
we could have safety session toggle restoring previous behavior

Reply:
Are there any use cases where presto defined type differs from parquet types? (I know about logical types there are not fully supported)

Reply:
Generally -- any time table schema changes (schema evolution). So we will add conversions here too.

Reply:
Do we have a matrix of type-to-type conversions that are possible (Parquet <-> Presto)?